### PR TITLE
Use stable nixpkgs version (23.11)

### PR DESCRIPTION
**Pull request by georgeee**
Problem: some old unstable nixpkgs is used in flake.nix, which is potentially insecure.

Solution: bump the version of nixpkgs, and a few more flakes.

Explain how you tested your changes:
* [x] Test build on MacOS Arm
* [x] Test build on MacOS x86
* [x] Test build on Linux

Checklist:

- [x] Dependency versions are unchanged
  - Notify Velocity team if dependencies must change in CI
- [x] Modified the current draft of release notes with details on what is completed or incomplete within this project
- [x] Document code purpose, how to use it
  - Mention expected invariants, implicit constraints
- [x] Tests were added for the new behavior
  - Document test purpose, significance of failures
  - Test names should reflect their purpose
- [x] All tests pass (CI will check this if you didn't)
- [x] Serialized types are in stable-versioned modules
- [x] Does this close issues? None

### CODE REVIEW

**Comment by georgeee** [comment]
!ci-build-me

**Comment by georgeee** [comment]
!ci-build-me

**Comment by georgeee** [comment]
!ci-build-me

**Comment by georgeee** [comment]
!ci-build-me

**Comment by georgeee** [comment]
!ci-build-me

**Comment by L-as** [comment]
Why not use unstable?

**Comment by georgeee** [comment]
!ci-build-me

**Comment by georgeee** [comment]
!ci-build-me

**Comment by georgeee** [comment]
!ci-build-me

**Comment by rbonichon** [comment]
Can confirm it works on NixOS as is. Have some issues on MacOS. Waiting for others with Apple Silicon to chime in to see if it's local or more global.

File: flake.nix [hunk 1]
```diff
@@ -261,6 +261,11 @@
         };
     } // utils.lib.eachDefaultSystem (system:
       let
+	rocksdbOverlay = pkgs: prev:
+          if builtins.elem system [ "aarch64-darwin" "x86_64-darwin" ] then
```
Comment (joaosreis) [inline flake.nix:265]: This change is an optional suggestion
```suggestion
          if prev.stdenv.isDarwin then
```

**Review by joaosreis — CHANGES_REQUESTED** [review]
Still has issues building in M1 macOS.

File: flake.nix [hunk 2]
```diff
@@ -261,6 +261,11 @@
...
+            { rockdb-mina = pkgs.rocksdb; }
```
[the remark this answers is not in the document]
Comment (georgeee) [reply flake.nix:266]: Oh, that's a pity. Don't know why I didn't catch it when testing before the commit.

**Comment by georgeee** [comment]
!ci-build-me

**Comment by georgeee** [comment]
!ci-toolchain-me

**Comment by georgeee** [comment]
!ci-build-me

**Comment by georgeee** [comment]
!ci-build-me

**Review by joaosreis — APPROVED** [review]
It's working well for me! (ARM macOS)

**Review by dkijania — APPROVED** [review]

**Comment by rbonichon** [comment]
Working well for me too!

**Review by rbonichon — APPROVED** [review]
Works fine. Thanks for the effort @georgeee !

**Comment by nholland94** [comment]
!approved-for-mainnet

**Pull request merged**